Kudu.Core/SourceControl/Git/GitExeRepository.cs

```diff
@@ -367,19 +367,28 @@ public void ClearLock()
         {
             // Delete the lock file from the .git folder
             var lockFilesPath = Path.Combine(_gitExe.WorkingDirectory, ".git");
+            // Always clear lock file under ".git/refs/heads"
+            var branchLockFiles = Path.Combine(_gitExe.WorkingDirectory, ".git", "refs", "heads");
 
             if (!Directory.Exists(lockFilesPath))
             {
                 return;
             }
 
-            var lockFiles = Directory.EnumerateFiles(lockFilesPath, "*.lock", SearchOption.AllDirectories)
+            List<string> lockFiles = Directory.EnumerateFiles(lockFilesPath, "*.lock", SearchOption.AllDirectories)
```

Comment:
@davidebbo 

looked at the [original issue](https://github.com/projectkudu/kudu/issues/634), there will be multiple HEAD.lock in different folder (module folder), so we will have to go thru all folder recursively.


@ahmelsayed 
DirectoryBase from IFileSystem doesn`t have EnumerateFiles API, i think we can implement it, but the existing test case is running against real file system, we are fine with testing. i will leave it as it is for now, we can add it later if we really want to mock the file system.